### ISSUE TITLE
Panicked at 'Box<Any>', compiler\rustc_errors\src\lib.rs:945:9

### ISSUE THREAD

**Issue by DevQps** [rust-lang/rust-clippy#7158]
# Description
I was compiling my code using nightly rust (1.51) and suddenly received the following error today.
Hopefully the copied message helps finding the issue. Along with this message I also received some additional messages (normal compiler errors, which I have removed from here)

```
error: internal compiler error: compiler\rustc_middle\src\ty\sty.rs:2150:18: tuple_fields called on non-tuple

thread 'rustc' panicked at 'Box<Any>', compiler\rustc_errors\src\lib.rs:945:9
stack backtrace:
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust-clippy/issues/new

note: Clippy version: clippy 0.0.212 (ffa2e7ae8 2020-10-24)

query stack during panic:
#0 [typeck] type-checking `create_tcp_client`
#1 [mir_built] building MIR for `create_tcp_client`
#2 [unsafety_check_result] unsafety-checking `create_tcp_client`
#3 [mir_const] processing MIR for `create_tcp_client`
#4 [mir_promoted] processing `create_tcp_client`
#5 [mir_borrowck] borrow-checking `create_tcp_client`
#6 [type_of] computing type of `create_tcp_client::{opaque#0}`
#7 [check_mod_item_types] checking item types in top-level module
#8 [analysis] running analysis passes on this crate
end of query stack
error: aborting due to 4 previous errors
```

**Comment by anall** [comment]
Can you please run `clippy` with `RUST_BACKTRACE=full` and provide the backtrace.

If possible, can you also provide a minimal reproduction for the issue or, failing that, the source code?

Thanks.

**Comment by giraffate** [comment]
Thanks for the report!

I think this is the same case of https://github.com/rust-lang/rust/issues/77993 and seems to be fixed now (FYI 1.51 is currently stable). So I'm closing this. If you have anything, feel free to reopen this.
